synapse/cores/common.py

```diff
@@ -2600,8 +2600,9 @@ def _tufosByInetCidr(self, prop, valu, limit=None):
 
         ipv4str, cidr = valu.split('/', 1)
         ipv4addr, _ = s_datamodel.getTypeParse('inet:ipv4', ipv4str)
-        mask = (2 ** (32 - int(cidr)))
-        ipv4addr &= ~mask
+        shift = 32 - int(cidr)
+        ipv4addr = ipv4addr >> shift << shift
+        mask = (2 ** shift)
```

Comment:
it's a bit of a perf nit, but we should be able to use int(cidr) to lookup 2 mask values from a list to avoid dynamic calculation of the values.  For example, /24 offset would return:
```
(0xffffff00, 0xff )
```
And we would do something like:
```
minaddr = ipv4addr & mask[0]
maxaddr = minaddr + mask[1] 
``` 
it's less relevant in python, but "and" operations are faster than shift operations. :D